lib/core/cursor.js

```diff
@@ -435,7 +435,7 @@ class CoreCursor extends Readable {
           return;
         }
 
-        cursor._next(callback);
+        this._initializeCursor(callback);
```

Comment:
What is the change happening here?

Reply:
This was that last isolated commit. Basically, if a session check is needed then we do that, but haven't necessarily initialized the cursor (this is called within `initializeCursor`). Sometimes this _just works_ because `next` figures out that the cursor isn't initialized yet, but there were cases (mostly unit tests with the mock sever) where state was set inside the cursor such that it was immediately issuing a `getMore` instead. I think calling `initializeCursor` here is strictly more correct